include/aspect/mesh_deformation/free_surface.h

```diff
@@ -130,7 +130,7 @@ namespace aspect
          */
         struct SurfaceAdvection
         {
-          enum Direction { normal, vertical };
+          enum Direction { normal, vertical, geometric_normal };
```

Comment:
Can you document what these flags mean?

Reply:
Separately, why are the existing options not good enough?

Reply:
I think this is still not resolved?

Reply:
Ah, I had removed all other instances of `geometric_normal`, except for this one. It's gone now. I have not added documentation, as I don't think that should be part of this PR, which is only about the diffusion plugin.